### PR TITLE
Convert a bunch of small files to JS

### PR DESCRIPTION
**Pull request by mnquintana**
Continuing on from https://github.com/atom/atom/pull/13065, this PR rewrites a bunch of small files as JS. 🤘 

👀 @atom/feedback

### CODE REVIEW

File: src/model.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    nextInstanceId = 1
+  }
+
+  constructor (params = {}) {
+    this.alive = true
+    this.assignId(params.id)
+  }
+
+  assignId (id) {
+    this.id = this.id || id || nextInstanceId++
```
Comment (mnquintana) [inline src/model.js]: I don't think it will since we initialize it to `1`, but that's a good point that this could be more robust

**Comment by mnquintana** [comment]
Huh, looks like `atomdoc` is blowing up on some unknown docstring 😞  – I'll take a look into adding more debug info to the build script to make it easier to find which file has invalid Atomdoc going on 🕵️

**Comment by joh1234** [comment]
Great, thanks. https://goo.gl/JBljdR

**Comment by mnquintana** [comment]
Sat on this one a bit too long, and I'm preeeetty sure at least some of these files have seen some edits since I opened this PR. Gonna do a new one with fewer files.

**Pull request closed without merging**